tests/workflows/basic_emitter_workflow/tests/test_workflow.py

```diff
@@ -51,179 +52,109 @@ def test_run_workflow__happy_path(mock_uuid4_generator, mock_datetime_now):
-                        f"{base_module}.workflow.NextNode": [],
-                    },
-                    "dependencies_invoked": {
-                        str(next_node_span_id): [f"{base_module}.workflow.StartNode"],
-                    },
+                "node_executions_initiated": {
+                    f"{base_module}.workflow.StartNode": [str(start_node_span_id)],
+                    f"{base_module}.workflow.NextNode": [str(next_node_span_id)],
                 },
-                "parent": None,
-            },
-            "score": 13,
-        },
-    )
-
-    assert not DeepDiff(
-        state_snapshots[4],
-        {
-            "meta": {
-                "id": str(state_id),
-                "trace_id": str(trace_id),
-                "span_id": str(workflow_span_id),
-                "updated_ts": "2024-01-01T12:00:00",
-                "is_terminated": True,
-                "workflow_inputs": {},
-                "external_inputs": {},
-                "node_outputs": {
-                    "StartNode.Outputs.final_value": "Hello, World!",
-                    "NextNode.Outputs.final_value": "Score: 13",
+                "node_executions_queued": {
+                    f"{base_module}.workflow.NextNode": [],
                 },
-                "node_execution_cache": {
-                    "node_executions_fulfilled": {
-                        f"{base_module}.workflow.StartNode": [str(start_node_span_id)],
-                        f"{base_module}.workflow.NextNode": [str(next_node_span_id)],
-                    },
-                    "node_executions_initiated": {
-                        f"{base_module}.workflow.StartNode": [str(start_node_span_id)],
-                        f"{base_module}.workflow.NextNode": [str(next_node_span_id)],
-                    },
-                    "node_executions_queued": {
-                        f"{base_module}.workflow.NextNode": [],
-                    },
-                    "dependencies_invoked": {
-                        str(next_node_span_id): [f"{base_module}.workflow.StartNode"],
-                    },
+                "dependencies_invoked": {
+                    str(next_node_span_id): [f"{base_module}.workflow.StartNode"],
                 },
-                "parent": None,
             },
-            "score": 13,
+            "parent": None,
         },
-    )
+        "score": 13,
+    }
 
-    assert len(state_snapshots) == 5
+    assert events[7].name == "node.execution.fulfilled"
+    assert events[7].node_definition == NextNode
+    assert events[7].outputs == {"final_value": "Score: 13"}
+
+    assert events[8].name == "workflow.execution.fulfilled"
+    assert events[8].outputs == {"final_value": "Score: 13"}
+
+    assert len(events) == 9, final_event
+
+    # AND the emitter should have emitted all of the expected state snapshots
+    state_snapshots = list(emitter.state_snapshots)
+    assert len(state_snapshots) == 3
```

Comment:
I'm also currently planning to remove the snapshot_state method from emitters now that state can be emitted through events